File: eval/embed.py
"""OpenAI embeddings for the eval harness (mirrors the agent's embed model)."""
import os

from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()

_client = OpenAI(api_key=os.environ.get("OPENAI_API_KEY"))
MODEL = os.environ.get("EMBEDDING_MODEL", "text-embedding-3-small")


def embed(texts: list[str]) -> list[list[float]]:
    if not texts:
        return []
    res = _client.embeddings.create(model=MODEL, input=texts)
    return [d.embedding for d in res.data]

File: src/ingest/embed.ts
import OpenAI from "openai";
import { config } from "../config.js";

const client = new OpenAI({ apiKey: config.openaiApiKey });

/** Batch-embed texts with the configured model (text-embedding-3-small, 1536-d). */
export async function embedBatch(texts: string[]): Promise<number[][]> {
  if (texts.length === 0) return [];
  if (!config.openaiApiKey) {
    throw new Error("OPENAI_API_KEY is required for material embeddings.");
  }
  const res = await client.embeddings.create({
    model: config.embeddingModel,
    input: texts,
  });
  return res.data.map((d) => d.embedding as number[]);
}
